.github/workflows/pr-metadata-checks.yml

```diff
@@ -2,13 +2,19 @@ name: Check Pull Request Metadata
 on:
   pull_request_target:
     # trigger when the PR title changes
-    types: [opened, edited, reopened]
+    types: [opened, synchronize, reopened, edited]
 
 jobs:
-  pr-title:
+  check-pr-metadata:
     runs-on: ubuntu-22.04
     permissions:
+      contents: read
       pull-requests: write # post comments when the PR title doesn't match the "Conventional Commits" rules
     steps:
       - name: Check Pull Request title
+        if: contains(fromJSON('["opened", "edited", "reopened"]'), github.event.action)
         uses: bonitasoft/actions/packages/pr-title-conventional-commits@v3
+
+      - name: Add labels to PR
```

Comment:
**suggestion:** this could be put in a dedicated job because if the first step fails, the 2nd one doesn't run.